modules/apps/jax-rs/mdm-admin/src/main/java/org/wso2/carbon/mdm/api/util/CredentialManagementResponseBuilder.java

```diff
@@ -0,0 +1,105 @@
+     * @param credentials - User credentials
+     * @return Response Object
+     * @throws MDMAPIException
+     */
+    public static Response buildChangePasswordResponse(UserCredentialWrapper credentials) throws MDMAPIException {
+        UserStoreManager userStoreManager = MDMAPIUtils.getUserStoreManager();
+        ResponsePayload responsePayload = new ResponsePayload();
+
+        try {
+            byte[] decodedNewPassword = Base64.decodeBase64(credentials.getNewPassword());
+            byte[] decodedOldPassword = Base64.decodeBase64(credentials.getOldPassword());
+            userStoreManager.updateCredential(credentials.getUsername(), new String(
+                    decodedNewPassword, "UTF-8"), new String(decodedOldPassword, "UTF-8"));
+            responsePayload.setStatusCode(HttpStatus.SC_CREATED);
+            responsePayload.setMessageFromServer("User password by username: " + credentials.getUsername() +
+                    " was successfully changed.");
+            return Response.status(HttpStatus.SC_CREATED).entity(responsePayload).build();
+        } catch (UserStoreException e) {
+            String errorMsg = "Exception in trying to change the password by username: " + credentials.getUsername();
+            log.error(errorMsg, e);
+            responsePayload.setStatusCode(HttpStatus.SC_BAD_REQUEST);
+            responsePayload.setMessageFromServer("Old password does not match.");
+            return Response.status(HttpStatus.SC_BAD_REQUEST).entity(responsePayload).build();
+        } catch (UnsupportedEncodingException e) {
+            String errorMsg = "Exception in trying to change the password by username: " + credentials.getUsername();
+            log.error(errorMsg, e);
+            throw new MDMAPIException(errorMsg, e);
+        }
+
+    }
+
+    /**
+     * Builds the response to reset the password of a user
+     * @param credentials - User credentials
+     * @return Response Object
+     * @throws MDMAPIException
+     */
+    public static Response buildResetPasswordResponse(UserCredentialWrapper credentials) throws MDMAPIException {
+        UserStoreManager userStoreManager = MDMAPIUtils.getUserStoreManager();
+        ResponsePayload responsePayload = new ResponsePayload();
+        try {
+            byte[] decodedNewPassword = Base64.decodeBase64(credentials.getNewPassword());
+            userStoreManager.updateCredentialByAdmin(credentials.getUsername(), new String(
+                    decodedNewPassword, "UTF-8"));
+            responsePayload.setStatusCode(HttpStatus.SC_CREATED);
+            responsePayload.setMessageFromServer("User password by username: " + credentials.getUsername() +
+                    " was successfully changed.");
+            return Response.status(HttpStatus.SC_CREATED).entity(responsePayload).build();
+        } catch (UserStoreException e) {
+            String errorMsg = "Exception in trying to change the password by username: " + credentials.getUsername();
```

Comment:
better explaining the specific error message here for the cause of exception rather than putting a generic message.